### PR TITLE
Remove daskexcutor provider from Airflow code

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: .github/boring-cyborg.yml [hunk 1]
```diff
@@ -183,21 +183,16 @@ labelPRBasedOnFilePath:
...
   provider:databricks:
     - airflow/providers/databricks/**/*
     - docs/apache-airflow-providers-databricks/**/*
     - tests/providers/databricks/**/*
     - tests/system/providers/databricks/**/*
 
   provider:datadog:
-    - airflow/providers/daskexecutor/**/*
-    - docs/apache-airflow-providers-daskexecutor/**/*
-    - tests/providers/daskexecutor/**/*
```
Comment (hussein-awala) [inline .github/boring-cyborg.yml:200]: Oups!

**Review by eladkal — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]
RIP

File: docs/spelling_wordlist.txt [hunk 2]
```diff
@@ -359,7 +359,6 @@ DagRunState
 DAGs
 Dask
 dask
```
Comment (eladkal) [inline docs/spelling_wordlist.txt]: can we remove these too?

File: scripts/ci/pre_commit/pre_commit_insert_extras.py [hunk 3]
```diff
@@ -45,13 +45,6 @@
 CONSTANTS_HEADER = "# START EXTRAS HERE"
 CONSTANTS_FOOTER = "# END EXTRAS HERE"
 
-DEFAULT_EXTRAS = (
```
Comment (potiuk) [inline scripts/ci/pre_commit/pre_commit_insert_extras.py:48]: This turned out to be not used any more.

File: docs/spelling_wordlist.txt (hunk 2 above)
Comment (potiuk) [reply docs/spelling_wordlist.txt]: I am not sure if we can even remove daskexecutor to be honest (due to release notes),

File: .github/boring-cyborg.yml (hunk 1 above)
Comment (potiuk) [reply .github/boring-cyborg.yml:200]: Indeed :)

File: docs/spelling_wordlist.txt (hunk 2 above)
Comment (potiuk) [reply docs/spelling_wordlist.txt]: Yeah. We can't even remove `daskexecutor` because of release notes.

**Review by josh-fell — APPROVED** [review]

**Comment by potiuk** [comment]
Booom!

**Pull request merged**